### PR TITLE
Use new actions-lib for ci

### PR DESCRIPTION
**Pull request by joakimen**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,77 +6,76 @@ on:
...
+        with:
+          tag-prefix: "${{ steps.prep.outputs.major-version }}"
+          tag-type: "punctuated-timestamp-tag"
+
+      - name: resolve dependencies
+        env:
+          # NOTE: Use the following token if your project does not depend on private packages:
+          # GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # NOTE: Use the following token if your project depends on private packages:
+          GITHUB_TOKEN: ${{ secrets.SHARED_GITHUB_PACKAGES_TOKEN }}
```
Comment (stekern) [inline .github/workflows/ci.yml:64]: Not needed - the built-in GITHUB_TOKEN should suffice.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -6,77 +6,76 @@ on:
...
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          export MAVEN_OPTS="-Duser.home=$HOME" # Home directory not automatically picked up for some reason
-          major_version="$(sed -n "s/^.*<major-version>\([0-9]\{1,\}\)<\/major-version>.*$/\1/p" pom.xml)"
-          date="$(date -u "+%Y%m%d.%H%M%S")"
-          tag="$major_version.$date"
-          printf "Tag to use for potential release: %s\n" "$tag"
-          mvn -B org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce -Drules=requireReleaseDeps
+          SONAR_TOKEN: ${{ secrets.SHARED_SONAR_TOKEN }}
```
Comment (stekern) [inline .github/workflows/ci.yml:70]: SonarCloud isn't currently configured for this repository, so I suggest we remove this for now

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -6,77 +6,76 @@ on:
...
-            # Release to GitHub Packages
-            mvn -B source:jar deploy scm:tag -DaltReleaseDeploymentRepository="github::default::https://maven.pkg.github.com/capralifecycle/liflig-properties" -Drevision="$tag" -Dtag="$tag"
+        run: |
+          if [ "$CONDITIONAL_RELEASE" = "true" ]; then
+            echo "Releasing library with tag '$TAG'"
+            mvn -B source:jar deploy scm:tag -Drevision="$TAG" -Dtag="$TAG" \
+              org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce -Drules=requireReleaseDeps
           else
-            mvn -B source:jar verify -Drevision="$tag" -Dtag="$tag"
+            mvn -B -U verify
```
Comment (stekern) [inline .github/workflows/ci.yml:80]: Do we want to run `source:jar` here as well?

**Review by stekern — COMMENTED** [review]
Some minor comments, but besides that this looks good 🚀

File: .github/workflows/ci.yml (hunk 3 above)
Comment (joakimen) [reply .github/workflows/ci.yml:80]: I don't `source:jar` will help provoke any more potential configuration issues, so considering we don't deploy the source/app jars from this stage I don't think we need it. Compile/test should be fine

File: .github/workflows/ci.yml (hunk 1 above)
Comment (joakimen) [reply .github/workflows/ci.yml]: https://github.com/capralifecycle/liflig-properties/pull/57/commits/0c2cc46e464f194d04c9719e192b811c76f86e40

File: .github/workflows/ci.yml (hunk 2 above)
Comment (joakimen) [reply .github/workflows/ci.yml]: https://github.com/capralifecycle/liflig-properties/pull/57/commits/8931fdaa803e68be86832c15d6528b7a87510012

**Review by stekern — APPROVED** [review]
LGTM 👌

**Pull request merged**